src/Aspire.Hosting.Azure/Provisioning/Provisioners/AzureProvisioner.cs

```diff
@@ -362,14 +364,44 @@ await foreach (var tenant in armClient.GetTenants().GetAllAsync(cancellationToke
             throw new MissingConfigurationException("An azure location/region is required. Set the Azure:Location configuration value.");
         }
 
-        var unique = $"{Environment.MachineName.ToLowerInvariant()}-{environment.ApplicationName.ToLowerInvariant()}";
+        var userSecrets = await userSecretsLazy.Value.ConfigureAwait(false);
+
+        string resourceGroupName;
+        bool createIfAbsent;
 
-        // Name of the resource group to create based on the machine name and application name
-        var (resourceGroupName, createIfAbsent) = _options.ResourceGroup switch
+        if (string.IsNullOrEmpty(_options.ResourceGroup))
         {
-            null or { Length: 0 } => ($"rg-aspire-{unique}", true),
-            string rg => (rg, _options.AllowResourceGroupCreation ?? false)
-        };
+            // Generate an resource group name since none was provided
+
+            var prefix = "rg-aspire";
+
+            if (!string.IsNullOrWhiteSpace(_options.ResourceGroupPrefix))
+            {
+                prefix = _options.ResourceGroupPrefix;
+            }
+
+            var suffix = RandomNumberGenerator.GetHexString(8, lowercase: true);
+
+            var maxApplicationNameSize = ResourceGroupNameHelpers.MaxResourceGroupNameLength - prefix.Length - suffix.Length - 2; // extra '-'s
+
+            var normalizedApplicationName = ResourceGroupNameHelpers.NormalizeResourceGroupNameForAzd(environment.ApplicationName.ToLowerInvariant());
```

Comment:
Why do we ToLower the ApplicationName? That makes it harder to read IMO because I don't get PascalCasing anymore.

Reply:
Was like this before the PR. If you get support or strongly think so then I'll change it.
NB: In the portal they are case-insensitive if that helps deciding we can keep the original casing.

Reply:
@davidfowl - is there a reason for this? Seems like it has negative value IMO.